src-tauri/src/main.rs

```diff
@@ -415,18 +415,22 @@ async fn run_algorithms(
     };
 
     Ok(APMincutStringResults {
-        ap_result: ap_string_vec,
-        mincut_result: mincut_string_vec,
+        ap_result: ap_vec,
+        mincut_result: mincut_vec,
     })
 }
 
 pub fn node_index_to_key(
     nodeindex: &petgraph::graph::NodeIndex,
     graph: &graph::graph_ds::Graph,
-) -> Option<String> {
+) -> Option<u32> {
     graph.node_idx_map.iter().find_map(|(key, &val)| {
         if val == *nodeindex {
-            Some(key.clone())
+            let num = key.parse::<u32>();
+            match num {
```

Comment:
if there's a better way to do this, let me know

Reply:
Why wouldn't you directly return `num` if it's of type `Option<u32>`?

Reply:
unwrapped w ok